dockstore-webservice/src/main/java/io/dockstore/webservice/helpers/ZenodoHelper.java

```diff
@@ -162,33 +174,16 @@ public static void registerZenodoDOIForWorkflow(ApiClient zenodoClient, String d
 
         workflow.setConceptDoi(conceptDoi);
 
+        // Only add the alias to the workflow after publishing the DOI succeeds
+        // Otherwise if the publish call fails we will have added an alias
+        // that will not be used and cannot be deleted
+        // This code also checks that the alias does not start with an invalid prefix
+        // If it does, this will generate an exception, the alias will not be added
+        // to the workflow, but there may be an invalid Related Identifier URL on the Zenodo entry
+        workflowResource.getEntryResource().addAliasesAndCheck(user, workflow.getId(), doiAlias, false);
```

Comment:
Additionally, rather than modify workflowResource as a side effect, I note that the method is void. 
It would be more understandable to make these changes from the return value.